### DISCUSSION TITLE
how to get detail Type included type of field

### DISCUSSION THREAD

**Discussion by ardani** [webonyx/graphql-php#1161] [Q&A]
Hi Guys,
do you know how to get detail Type from schema?
for example, we define schema like this

```
type TestCredential {
    service: String
    category: String
    label: String
    identifier: String
}
```
how to get type TestCredential Type included field
expected result 
```
$type = [
'TestCredential' => [
   'service' => 'String'
   ...etc
  ]
];
```

thanks!

**Comment by spawnia** [comment]
Use `GraphQL\Language\Parser`.

**Reply by ardani** [reply]
thanks for the answer
@spawnia do you know how to use it in resolver? is it possible to use Parser::parse data from ResolveInfo?

**Reply by spawnia** [reply]
It is very unclear what you are actually trying to do. You original question asked about getting a programmatically accessible representation of a schema string. What do you need to do?

**Reply by ardani** [reply]
@spawnia sorry if my question unclear.
1.  I'm implementing graphql but I created graphql schema file first (main.graphql) then I loaded it to BuildSchema
2. I setup global resolver for that and set setDefaultFieldResolver
3. in that function I want to get information Type TestCredential above from schema included type of field so I can format the response as we define in type.
4. is it possible to parse Type TestCredential in function setDefaultFieldResolver? in that function we have these parameters
![image](https://user-images.githubusercontent.com/2499777/171671776-6b0ad554-d847-4241-948a-884444a48d25.png)

**Reply by spawnia** [reply]
`ResolveInfo::$returnType` contains the type that is returned from a field.

Threading everything through the default field resolver seems very wrong though. Prefer defining resolvers per field.
